src/main/java/org/broadinstitute/consent/http/models/support/SupportRequest.java

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.consent.http.models.support;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.broadinstitute.consent.http.enumeration.SupportRequestType;
+
+/**
+ * Represents Request object for the Zendesk API
+ * <a href="https://developer.zendesk.com/api-reference/ticketing/tickets/ticket-requests/">Requests API Reference</a>
+ */
+public class SupportRequest {
```

Comment:
Could this also be a record? It looks like the only use of setters here is from test code, and those cases could be supported by adding additional constructors.

Here's what I have in mind. The test would have to be changed to call the new constructor to initialize the comment and custom field properties.
```
public record SupportRequest(SupportRequester requester, String subject, List<CustomRequestField> customFields,
    SupportRequestComment comment, long ticketFormId) {

  public SupportRequest(String name, SupportRequestType type, String email, String subject,
      String description, String url, List<String> uploads) {
    this(new SupportRequester(name, email), subject, createCustomFields(name, type, email, description),
        new SupportRequestComment(description + "\n\n------------------\nSubmitted from: " + url, uploads));
  }

  public SupportRequest(SupportRequester supportRequester, String subject,
      List<CustomRequestField> customFields, SupportRequestComment supportRequestComment) {
    this(supportRequester, subject, customFields, supportRequestComment, 360000669472L);
  }

  private static List<CustomRequestField> createCustomFields(String name, SupportRequestType type,
      String email, String description) {
    return List.of(new CustomRequestField(360012744452L, type.getValue()),
        new CustomRequestField(360007369412L, description),
        new CustomRequestField(360012744292L, name),
        new CustomRequestField(360012782111L, email),
        new CustomRequestField(360018545031L, email));
  }
}
```

Reply:
I like the idea ... will give it a shot.